hnn_core/tests/test_extracellular.py

```diff
@@ -0,0 +1,341 @@
+    for seg_idx in range(sec.nseg):
+        # PSA: distance to middle segment == electrode x-position
+        var_r_psa = np.sqrt(elec_pos[0] ** 2 +
+                            (elec_pos[1] - seg_ctr_pts[seg_idx]) ** 2)
+        target_vals['psa'].append(
+            1000 / (4. * np.pi * conductivity * var_r_psa))
+
+        # LSA: calculate L and H variables relative to segment endpoints
+        var_l = elec_pos[1] - (seg_ctr_pts[seg_idx] - seg_lens[seg_idx])
+        var_h = elec_pos[1] - (seg_ctr_pts[seg_idx] + seg_lens[seg_idx])
+        var_r_lsa = elec_pos[0]  # just use the axial distance
+        target_vals['lsa'].append(
+            1000 * np.log(np.abs(
+                (np.sqrt(var_h ** 2 + var_r_lsa ** 2) - var_h) /
+                (np.sqrt(var_l ** 2 + var_r_lsa ** 2) - var_l)
+            )) / (4. * np.pi * conductivity * 2 * seg_lens[seg_idx]))
+
+    for method in ['psa', 'lsa']:
+        res = _transfer_resistance(sec, elec_pos, conductivity, method)
+        assert_allclose(res, target_vals[method])
+
+
+@requires_mpi4py
+@requires_psutil
+def test_extracellular_backends(run_hnn_core_fixture):
+    """Test extracellular outputs across backends."""
+
+    electrode_array = {'arr1': [(2, 2, 400), (6, 6, 800)]}
+    _, joblib_net = run_hnn_core_fixture(
+        backend='joblib', n_jobs=1, reduced=True, record_isoma=True,
+        record_vsoma=True, electrode_array=electrode_array)
+    _, mpi_net = run_hnn_core_fixture(
+        backend='mpi', n_procs=2, reduced=True, record_isoma=True,
+        record_vsoma=True, electrode_array=electrode_array)
+
+    assert (len(electrode_array['arr1']) ==
+            len(joblib_net.rec_array['arr1'].positions) ==
+            len(mpi_net.rec_array['arr1'].positions))
+    assert (len(joblib_net.rec_array['arr1']) ==
+            len(mpi_net.rec_array['arr1']) ==
+            2)  # length == n.o. trials
+
+    # reduced simulation has n_trials=2
+    # trial_idx, n_trials = 0, 2
+    for tr_idx, el_idx in zip([0, 1], [0, 1]):
+        assert_allclose(joblib_net.rec_array['arr1']._data[tr_idx][el_idx],
+                        mpi_net.rec_array['arr1']._data[tr_idx][el_idx])
+
+    assert isinstance(joblib_net.rec_array['arr1'].get_data(), np.ndarray)
+    assert_array_equal(joblib_net.rec_array['arr1'].get_data().shape,
+                       [len(joblib_net.rec_array['arr1']._data),
+                        len(joblib_net.rec_array['arr1']._data[0]),
+                        len(joblib_net.rec_array['arr1']._data[0][0])])
+
+    # make sure sampling rate is fixed (raises RuntimeError if not)
+    _ = joblib_net.rec_array['arr1'].sfreq
+    # check plotting works
+    joblib_net.rec_array['arr1'].plot(show=False)
+
+
+def _mathematical_dipole(e_pos, d_pos, d_Q):
+    rr = e_pos - d_pos
+    R = norm(rr)
+    Q = norm(d_Q)
+    cosT = np.dot(rr, d_Q) / (R * Q)
+    return (Q * cosT) / (4 * np.pi * R ** 2)
+
+
+# require MPI to speed up due to large number of extracellular electrodes
```

Comment:
but this will not speed up in the CIs ...

Reply:
while I like this code, I think this is not really a "unit" test. Could we perhaps maintain this in your example repository?

Reply:
I'm not sure I know the definition, but this is clearly the slowest test. I guess I'm OK with cutting it out, will push update soon.